test/functional/clusteradm/clusteradmE2E_test.go

```diff
@@ -0,0 +1,252 @@
+			err = e2e.RunCmd(e2e.Streams, "accept", "--clusters", e2e.Values.ContextValue.ManagedCluster1.ClusterName, "--context", e2e.Values.ContextValue.Hub.ContextName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "clusteradm accept error")
+
+			ginkgo.By("get token from hub")
+			f, err := e2e.Create("token")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "create file error")
+
+			err = e2e.RunCmd(genericclioptions.IOStreams{Out: f}, "get", "token")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "clusteradm get token error")
+			f.Close()
+
+			ginkgo.By("managedcluster2 join hub")
+			tokenfilename := e2e.Values.DirectoryValue.ResultDirectory + "/token"
+			token, host, err = util.GetTokenHostHelper2(tokenfilename)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "read join command error")
+			err = e2e.RunCmd(e2e.Streams, "join",
+				"--context", e2e.Values.ContextValue.ManagedCluster2.ContextName,
+				"--hub-token", token, "--hub-apiserver", host,
+				"--wait",
+				"--cluster-name", e2e.Values.ContextValue.ManagedCluster2.ClusterName,
+			)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "managedcluster2 join error")
+
+			ginkgo.By("hub accept managedcluster2")
+			err = e2e.RunCmd(e2e.Streams, "accept", "--clusters", e2e.Values.ContextValue.ManagedCluster2.ClusterName, "--context", e2e.Values.ContextValue.Hub.ContextName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "clusteradm accept error")
+
+			ginkgo.By("delete token")
+			err = e2e.RunCmd(e2e.Streams, "delete", "token")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "clusteradm delete token error")
+
+			ginkgo.By("get token from hub")
+			b, err := ioutil.ReadFile(tokenfilename)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "read token file error")
+
+			r, w, _ := os.Pipe()
+			err = e2e.RunCmd(genericclioptions.IOStreams{Out: w}, "get", "token")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "clusteradm get token error")
+			w.Close()
+			newtoken, _ := ioutil.ReadAll(r)
+			gomega.Expect(newtoken).NotTo(gomega.Equal(b), "new token identical as previous token after delete")
+
+			// hub had accepted 2 mcls, you can write test code for other commands.
+			// TODO: write test code for other scenarios.
+		})
+
+	})
+
+})
+
+var _ = ginkgo.Describe("test clusteradm with bootstrap token", func() {
+	defer ginkgo.GinkgoRecover()
+
+	ginkgo.By("With Bootstrap Token")
+
+	ginkgo.BeforeEach(func(done ginkgo.Done) {
+		ginkgo.By("bootstrapping e2e test environment")
+		// get cluster infos
+		e2e2 = util.BootStrapE2eEnv("bootstrap")
+		close(done)
+	}, 600)
+
+	ginkgo.AfterEach(func() {
+		ginkgo.By("tearing down e2e test environment")
+		// clean
+		e2e2.KindDeleteCluster()
+	})
+
+	ginkgo.Context("join hub scenario", func() {
+		ginkgo.By("join hub with bootstrap token")
+
+		ginkgo.It("should managedcluster join hub successfully", func() {
+			ginkgo.By("clusteradm version check")
+			err := e2e2.RunCmd(e2e2.Streams, "version")
```

Comment:
e2e2.RunCmd("version")
Streams is the filed of e2e2, not need as input parameter.
use the name clusteradm("version")  may be better understand.

Reply:
The output will be redirected in some case, such as this: :https://github.com/open-cluster-management-io/clusteradm/pull/111/files#diff-d092bd45baec86399bbd7cf1c6a927c365a7db10ce559b3aca73e35604fce025R73 .
So i added the stream parameter. 
And I'm wondering whether i should remove this stream parameter and then define another function to handle these cases, or keep this parameter as it is?

And for the function name, i think your suggestion is really good, i will rename that function.